src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,304 +1,500 @@
-
-    haplotype_caller.addChild(annotate)
-    return annotate.rv()
+    input_bam = get_bam.rv(0)
+    input_bai = get_bam.rv(1)
+
+    if config.preprocess:
+        preprocess = job.wrapJobFn(run_germline_preprocessing,
+                                   config.cores,
+                                   get_bam.rv(0),
+                                   get_bam.rv(1),
+                                   config.genome_fasta,
+                                   config.genome_dict,
+                                   config.genome_fai,
+                                   config.phase,
+                                   config.mills,
+                                   config.dbsnp,
+                                   xmx=config.xmx).encapsulate()
+        get_bam.addChild(preprocess)
+
+        if config.preprocess_only:
+            output_dir = os.path.join(config.output_dir, uuid)
+            filename = '{}.{}.bam'.format(uuid, config.suffix)
+            output_bam = job.wrapJobFn(upload_or_move_job,
+                                       filename,
+                                       preprocess.rv(0),
+                                       output_dir)
+            preprocess.addChild(output_bam)
+            return preprocess.rv(0), preprocess.rv(1)
+        else:
+            input_bam = preprocess.rv(0)
+            input_bai = preprocess.rv(1)
+
+    hc_disk = PromisedRequirement(lambda x: 2 * x.size + human2bytes('5G'), input_bam)
+    haplotype_caller = job.wrapJobFn(gatk_haplotype_caller,
+                                     input_bam,
+                                     input_bai,
+                                     config,
+                                     cores=config.cores,
+                                     disk=hc_disk)
+
+    get_bam.addFollowOn(haplotype_caller)
+    output_dir = os.path.join(config.output_dir, uuid)
+    vqsr_name = '{}.raw{}.gvcf'.format(uuid, config.suffix)
+    output_gvcf = job.wrapJobFn(upload_or_move_job,
+                                vqsr_name,
+                                haplotype_caller.rv(),
+                                output_dir)
+    haplotype_caller.addChild(output_gvcf)
+    return haplotype_caller.rv()
 
 
 def download_shared_files(job, config):
     """
     Downloads reference files shared by all samples in the pipeline
 
-    :param config dict: pipeline configuration options
-    :return dict: updated config dictionary with shared fileStoreIDS
+    :param Namespace config: Pipeline configuration options
+    :return: Updated config with shared fileStoreIDS
+    :rtype: Namespace
     """
     job.fileStore.logToMaster('Downloading shared reference files')
-    reference_names = ['genome.fa', 'phase.vcf', 'mills.vcf', 'dbsnp.vcf', 'hapmap.vcf', 'omni.vcf']
-    if config['run_bwa']:
-        bwa_references = ['amb', 'ann', 'bwt', 'pac', 'sa']
-        reference_names.extend(bwa_references)
-    for name in reference_names:
+    references = {'genome_fasta'}
```

Comment:
Nit, but `references` seems like the wrong term here, since most of these files are indices or known variant files.